andela/Titan-Devs
# #163518687 Create social authentication for the application

#### What does this PR do?

Enable social authentication for the project .

#### Description of Task to be completed?

This PR create the following endpoints : 
 - /api/v1/auth/google : to enable social authentication via google 
- /api/v1/auth/facebook : to enable social authentication via facebook
-  /api/v1/auth/twitter : to enable social authentication via twitter 
#### How should this be manually tested
1.  Update your .env file with social authentication credentials CLIENT_ID and CLIENT_SECRET for all providers : 
 - For google follow [this guide](https://developers.google.com/adwords/api/docs/guides/authentication) 
 - For twitter follow [this guide ](https://themepacific.com/how-to-generate-api-key-consumer-token-access-key-for-twitter-oauth/994/)
- For facebook  [this guide ](https://goldplugins.com/documentation/wp-social-pro-documentation/how-to-get-an-app-id-and-secret-key-from-facebook/)

```
GOOGLE_CLIENT_ID= '*******'
GOOGLE_CLIENT_SECRET= '*******'
FACEBOOK_CLIENT_ID =  '*******'
FACEBOOK_CLIENT_SECRET =  '*******'
TWITTER_CLIENT_ID = '********'
TWITTER_CLIENT_SECRET = '*******'
```
2. Run the project and open each link in the description in your browser and check if you can login with your social networks.
#### Any background context you want to provide?
None  
#### What are the relevant pivotal tracker stories?
#163518687 
#### Questions:
Guys, If you know how this feature could be improved let me know, especially on the testing for social authentication with Twitter let me know.
